packages/schema-extract/src/file-linker.ts

```diff
@@ -118,84 +117,55 @@ export class SchemaLinker {
-    //                             res.enum = option.enum;
-    //                         } else {
-    //                             const enums = [];
-    //                             for (let i = 0; i < option.enum.length; i++) {
-    //                                 if (option.enum.includes(res.enum![i])) {
-    //                                     enums.push(option.enum[i]);
-    //                                 }
-    //                             }
-    //                             if (enums.length === 0) {
-    //                                 return {$ref: NeverId};
-    //                             } else {
-    //                                 res.enum = enums;
-    //                             }
-    //                         }
-    //                     }
-    //                 } else {
-    //                     return {$ref: NeverId};
-    //                 }
-    //             }
-    //             if (option.definedAt) {
-    //                 res.definedAt = option.definedAt;
-    //             }
-    //         }
-    //     }
-    //     return res;
-    // }
+    private handleIntersection(options: Schema[], schema: ModuleSchema, paramsMap?: Map<string, Schema>): Schema {
+        if (options.length === 0) {
+            return {$ref: UnknownId};
+        }
+        const first = {...options[0]};
+        let res = isRef(first) ? this.handleRef(first, schema, paramsMap) : first;
+        const rest = options.slice(1);
+        for (const o of rest) {
+            const option = isRef(o) ? this.handleRef(o, schema, paramsMap) : o;
+            if (res.type && option.type && res.type !== option.type) {
+                return {$ref: NeverId};
+            }
+            if (isObjectSchema(option) && (isInterfaceSchema(res) || isObjectSchema(res))) {
+                res = this.mergeObjects(res, option, schema, paramsMap);
+            }
+        }
+        return res;
+    }
+
+    private mergeObjects(object1: Schema & IObjectFields, object2: Schema & IObjectFields, schema: ModuleSchema, paramsMap?: Map<string, Schema>): Schema & IObjectFields {
+        if (!object1.properties || !object2.properties) {
+            return object1;
+        }
+        const res: Schema & IObjectFields & {properties: {[name: string]: Schema}} = {type: 'object', properties: {}};
+        for (const p in object1.properties) {
+            if (object1.properties.hasOwnProperty(p)) {
+                res.properties[p] = object1.properties[p];
+                if (object1.definedAt) {
+                    res.properties[p].definedAt = object1.definedAt;
+                }
+            }
+        }
+        for (const p in object2.properties) {
+            if (!res.properties.hasOwnProperty(p)) {
+                res.properties[p] = object2.properties[p];
+                if (object2.definedAt) {
+                    res.properties[p].definedAt = object2.definedAt;
+                }
+            } else {
```

Comment:
If this can only happen when we have an invalid intersection type, then the question is should we even try to handle invalid type, or should we leave to typescript to check that the code we're working with doesn't have invalid types.

Reply:
Okay, I changed handleIntersection to throw if we reach a "never" situation so we don't need to check for it here.